In [3]:
import pandas as pd

In [4]:
df = pd.read_csv('train (1).csv')

In [95]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [234]:
df_wom = df.query("Sex == 'female'")
df_wom_int = df.query("Sex == 'female'").shape[0]
df_wom_int

314

__*Женщин на борту было 314

In [254]:
sav_wom_int = df_wom.query("Survived == 1").shape[0]
sav_wom_int

233

__*Спаслось 233 женщины

In [255]:
i = round((sav_wom_int / df_wom_int) * 100, 0)
i

74.0

__*Спаслось 74 % женщин

In [236]:
df_man = df.query("Sex != 'female'")
df_man_int = df.query("Sex != 'female'").shape[0]
df_man_int

577

__*Мужчин на борту было 577

In [256]:
sav_man_int = df_man.query("Survived == 1").shape[0]
sav_man_int

109

__*Спаслось 109 мужчин

In [258]:
i = round((sav_man_int / df_man_int) * 100, 0)
i

19.0

__*Спаслось 19 % мужчин

__*Вывод: Мужчины отдавали свои места в лодках для спасения женщин.